src/main/java/teammates/ui/webapi/output/SessionResultsData.java

```diff
@@ -144,15 +147,14 @@ private static ResponseOutput buildSingleResponseForStudent(
         } else {
             // we don't want student to figure out who is who by using the hash
             recipientName = removeAnonymousHash(getRecipientNameOfResponse(response, bundle).getName());
+            recipientTeam = bundle.getRoster().getInfoForIdentifier(response.getRecipient()).getTeamName();
         }
 
         // process comments
         List<FeedbackResponseCommentAttributes> feedbackResponseComments =
                 bundle.getResponseCommentsMap().getOrDefault(response.getId(), Collections.emptyList());
         Queue<CommentOutput> comments = buildComments(feedbackResponseComments, bundle);
 
-        // Student does not need to know the teams for giver and/or recipient
-        // unless the student him/herself is the giver and/or recipient
```

Comment:
This might need some reviews as it might leak information

Reply:
Added a check that the recipient is not anonymous before adding the team name. Thus, there should not be any leaks as if the recipient is not anonymous there should not be any problem with knowing his team.